src/components/Avatar/Addon.js

```diff
@@ -0,0 +1,35 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import style from './style.css';
+
+const Addon = ({ position, offset, ...rest }) => {
+  const positions = useMemo(() => {
+    const p = position.split('-');
+    return {
+      [p[0]]: `${offset}px`,
```

Comment:
What if you just let the user override position with the `rest` prop in `Addon`?

Whitelisting just the 4 corner names could backfire. A future dev could get `top-right` and `right-top` confused.

It also feels a bit arbitrary to use the same offset for x and y. There could be cases where a consumer needs to alter the position a bit to accommodate for non-circular add-on content.

You could still provide a default of `{ bottom: '0', right: '0' }`

Reply:
The expectation is that it's always used in Avatar, which is always a circle. Happy to allow style to overwrite those settings. Also, i could whitelist the reversed names as well, the code will work for either. Curious what @mdespuits thinks about the approaches

Reply:
I feel like allowing overriding of arbitrary styles isn't really necessary for most cases. I like the idea of whitelisting the reverse options in case there is some confusion, but I tend to fall on the side of keeping small components like this closed and opening up changes as needed.

So unless there is a need for overriding styles instead of using CSS classes I'm fine with your approach so long as the others are whitelisted. However, there could be the argument that if they type it backwards they will get console proptype errors and it won't work as they expected, so they will probably figure out their error before it gets anywhere close to a live system.